### PR TITLE
Patch supply wrapper

### PR DESCRIPTION
**Pull request by thebevrishot**
## Implementation details
- Rename from x/supply to x/bank
- Patch logic and comment

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -257,8 +259,11 @@ func NewBandApp(
 	app.BankKeeper = bankkeeper.NewBaseKeeper(
 		appCodec, keys[banktypes.StoreKey], app.AccountKeeper, app.GetSubspace(banktypes.ModuleName), app.BlockedAddrs(),
 	)
+	// wrappedBankerKeeter overrides burn token behavior to instead transfer to community pool.
+	wrappedBankerKeeter := bandbank.NewWrappedBankKeeperBurnToCommunityPool(app.BankKeeper)
```
Comment (taobun) [inline app/app.go:263]: Keeper

File: app/app.go [hunk 2]
```diff
@@ -257,8 +259,11 @@ func NewBandApp(
...
+	wrappedBankerKeeter.SetAccountKeeper(app.AccountKeeper)
```
Comment (taobun) [inline app/app.go:264]: No need this function on bankKeeper.

File: app/app.go [hunk 3]
```diff
@@ -257,8 +259,11 @@ func NewBandApp(
 	app.BankKeeper = bankkeeper.NewBaseKeeper(
```
Comment (taobun) [inline app/app.go:259]: Can use wrap bank keeper as app.BankKeeper

**Review by taobun — CHANGES_REQUESTED** [review]

File: x/bank/keeper.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+// SetDistrKeeper sets distr module keeper for this WrappedBankKeeper instance.
+func (k *WrappedBankKeeper) SetDistrKeeper(dk *distrkeeper.Keeper) {
+	k.distrKeeper = dk
+}
+
+// SetAccountKeeper sets account module keeper for this WrappedBankKeeper instance.
+func (k *WrappedBankKeeper) SetAccountKeeper(ak banktypes.AccountKeeper) {
+	k.accountKeeper = ak
+}
```
Comment (taobun) [inline x/bank/keeper.go:47]: Remove this

**Review by taobun — CHANGES_REQUESTED** [review]

**Comment by taobun** [comment]
I have an opinion to create a new file `expected_keepers.go` to declare AccountKeeper and DistributeKeeper interface that we want to use. Like other modules should be a good way to do.

**Comment by thebevrishot** [comment]
@taobun Applied the idea, please review.

File: x/bank/keeper/keeper.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+// NewWrappedBankKeeperBurnToCommunityPool creates a new instance of WrappedBankKeeper
+// with its distrKeeper and accountKeeper members set to nil.
+func NewWrappedBankKeeperBurnToCommunityPool(bk bankkeeper.Keeper, acc types.AccountKeeper) *WrappedBankKeeper {
+	return &WrappedBankKeeper{bk, nil, acc}
+}
+
+// SetDistrKeeper sets distr module keeper for this WrappedBankKeeper instance.
+func (k *WrappedBankKeeper) SetDistrKeeper(dk *distrkeeper.Keeper) {
```
Comment (taobun) [inline x/bank/keeper/keeper.go:40]: receive types.DistributionKeeper

File: x/bank/keeper/keeper.go [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+type WrappedBankKeeper struct {
+	bankkeeper.Keeper
+
+	distrKeeper   types.DistributionKeeper
+	accountKeeper types.AccountKeeper
+}
+
+// NewWrappedBankKeeperBurnToCommunityPool creates a new instance of WrappedBankKeeper
+// with its distrKeeper and accountKeeper members set to nil.
+func NewWrappedBankKeeperBurnToCommunityPool(bk bankkeeper.Keeper, acc types.AccountKeeper) *WrappedBankKeeper {
```
Comment (taobun) [inline x/bank/keeper/keeper.go:35]: I think BankKeeper can return entire struct (base on their bankkeeper)

**Review by taobun — CHANGES_REQUESTED** [review]

File: app/app.go [hunk 7]
```diff
@@ -262,9 +263,14 @@ func NewBandApp(
 	app.AccountKeeper = authkeeper.NewAccountKeeper(
 		appCodec, keys[authtypes.StoreKey], app.GetSubspace(authtypes.ModuleName), authtypes.ProtoBaseAccount, maccPerms,
 	)
-	app.BankKeeper = bankkeeper.NewBaseKeeper(
-		appCodec, keys[banktypes.StoreKey], app.AccountKeeper, app.GetSubspace(banktypes.ModuleName), app.BlockedAddrs(),
+	// wrappedBankerKeeter overrides burn token behavior to instead transfer to community pool.
```
Comment (taobun) [inline app/app.go:266]: wrappedBankerKeeper

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**